typescript/sdk/src/metadata/ChainMetadataManager.ts

```diff
@@ -0,0 +1,272 @@
+import { Debugger, debug } from 'debug';
+
+import { exclude, isNumeric } from '@hyperlane-xyz/utils';
+
+import { chainMetadata as defaultChainMetadata } from '../consts/chainMetadata';
+import { ChainMap, ChainName } from '../types';
+
+import {
+  ChainMetadata,
+  getDomainId,
+  isValidChainMetadata,
+} from './chainMetadataTypes';
+
+export interface ChainMetadataManagerOptions {
+  loggerName?: string;
+}
+
+/**
+ * A set of utilities to manage chain metadata
+ * Validates metadata on construction and provides useful methods
+ * for interacting with the data
+ */
+export class ChainMetadataManager<MetaExt = any> {
```

Comment:
This is the subset of the code that used to live in MultiProvider code that handles the metadata. I broke it out so the MultiProtocolProvider could use it too